test/functional/rpc_rawtransaction.py

```diff
@@ -69,24 +74,129 @@ def setup_network(self):
         super().setup_network()
         self.connect_nodes(0, 2)
 
+    def sync_peers(self):
+        self.sync_mempools(self.nodes[0:3])
+        self.sync_blocks()
+
+    def generate_and_sync(self, node, blocks):
+        self.sync_peers()
+        self.nodes[node].generate(blocks)
+        self.sync_peers()
+
     def run_test(self):
-        self.log.info('prepare some coins for multiple *rawtransaction commands')
-        self.nodes[2].generate(1)
-        self.sync_all()
-        self.nodes[0].generate(COINBASE_MATURITY + 1)
-        self.sync_all()
-        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(),1.5)
-        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(),1.0)
-        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(),5.0)
-        self.sync_all()
-        self.nodes[0].generate(5)
-        self.sync_all()
-
-        self.log.info('Test getrawtransaction on genesis block coinbase returns an error')
+        self.log.info("Prepare some coins for multiple *rawtransaction commands")
+        self.generate_and_sync(node=2, blocks=1)
+        self.generate_and_sync(node=0, blocks=COINBASE_MATURITY + 1)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.5)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 1.0)
+        self.nodes[0].sendtoaddress(self.nodes[2].getnewaddress(), 5.0)
+        self.generate_and_sync(node=0, blocks=5)
+
+        self.getrawtransaction_tests()
+        self.createrawtransaction_tests()
+        self.sendrawtransaction_tests()
+        self.sendrawtransaction_testmempoolaccept_tests()
+        self.decoderawtransaction_tests()
+        self.transaction_version_number_tests()
+        if not self.options.descriptors:
+            self.raw_multisig_transaction_legacy_tests()
+
+    def getrawtransaction_tests(self):
+        addr = self.nodes[1].getnewaddress()
+        txid = self.nodes[0].sendtoaddress(addr, 10)
+        self.generate_and_sync(node=0, blocks=1)
+        vout = find_vout_for_address(self.nodes[1], txid, addr)
+        rawTx = self.nodes[1].createrawtransaction([{'txid': txid, 'vout': vout}], {self.nodes[1].getnewaddress(): 9.999})
+        rawTxSigned = self.nodes[1].signrawtransactionwithwallet(rawTx)
+        txId = self.nodes[1].sendrawtransaction(rawTxSigned['hex'])
+        self.generate_and_sync(node=0, blocks=1)
```

Comment:
I am not sure whether having two `txid` and `txId` transactions helps here, given both of them are confirmed. Because of that we are also not getting a test for only `mempool` transactions.

So maybe we can have a `txid` confirmed in a block, and then have `txId` in the mempool to check for both the scenario?

Will make the purpose of two transactions clear.

Also, the names are confusing, something more distinct would be helpful.

Reply:
Interesting.  We use and need all three transactions, AFAICT.  Updated the names in the last commit to txid, txid2, and txid3.  The test still passes if we sync_blocks() for the second and third transactions instead of sync_all(), but I'd rather not mess with this here.

Reply:
Took a fresh look this morning, good call!  The first conditional in `CTransactionRef GetTransaction` was not being exercised by the tests.
```c++
    if (mempool && !block_index) {
        CTransactionRef ptx = mempool->get(hash);
        if (ptx) return ptx;
    }
```
Added this assert that covers it:
```python
             # 1. valid parameters - supply txid along with various valid values for verbose
+            if n == 0 or n == 3:
+                # test with a tx in mempool, with and without -txindex
+                tx_in_mempool = self.nodes[n].sendtoaddress(self.nodes[n + 1].getnewaddress(), 0.1)
+                self.nodes[n].getrawtransaction(tx_in_mempool, True)
```

Reply:
(Verified by removing that code in GetTransaction() to ensure the new assert fails without it and passes with it.)